### PR TITLE
Make table writing always distributed

### PR DESCRIPTION
**Pull request by haozhun**
~~Depends on #2390~~

### CODE REVIEW

**Comment by cberner** [comment]
Second commit looks good, but change the commit message to "Make table writing always distributed"

**Comment by haozhun** [comment]
Depends on #2390

**Comment by cberner** [comment]
Can you rebase this to see if that fixes the Travis tests?

**Comment by cberner** [comment]
Closing this, since @haozhun says you're implementing this @martint

**Comment by cberner** [comment]
This needs to be rebased

**Comment by haozhun** [comment]
Depend on #2390. Ready for review.

**Comment by cberner** [comment]
@martint assign back to me, once you've reviewed #2390

**Comment by haozhun** [comment]
@cberner Dependency on #2390 has been removed because @electrum wants this in today. Please review.

File: presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java [hunk 1]
```diff
@@ -285,7 +286,20 @@ else if (fragment.getOutputPartitioning() == OutputPartitioning.HASH) {
...
-                buffers.put(taskId, new HashPagePartitionFunction(nodeIndex, parentTasks.size(), getPartitioningChannels(fragment), getHashChannel(fragment), fragment.getTypes()));
+                buffers.put(taskId, new HashPagePartitionFunction(nodeIndex, parentTasks.size(), getPartitioningChannels(fragment).get(), getHashChannel(fragment), fragment.getTypes()));
+            }
+
+            newOutputBuffers = startingOutputBuffers
+                    .withBuffers(buffers.build())
+                    .withNoMoreBufferIds();
+        }
+        else if (fragment.getOutputPartitioning() == OutputPartitioning.ROUND_ROBIN) {
+            checkArgument(noMoreParentNodes, "Hash partitioned output requires all parent nodes be added in a single call");
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java]: wrong error message

**Comment by cberner** [comment]
`query.initial-hash-partitions` now controls the number of partitions for arbitrary partitioning, so we should probably rename it. You can file an issue rather than putting that in this PR though

**Comment by cberner** [comment]
Had a few minor comments, also check that Travis passes. Looks good

**Pull request merged**